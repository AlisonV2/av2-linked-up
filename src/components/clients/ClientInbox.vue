<template>
  <div class="row client-inbox">
    <div class="col-12" v-for="message in messages" :key="message.project">
      <router-link
        :to="{ name: 'MessageDetails', params: { id: message.project } }"
      >
        <div class="message-item">Conversation with : {{ message.artist }}</div>
      </router-link>
    </div>
  </div>
</template>

<script>
/**
 * @exports ClientInbox
 * @type {Component}
 */
export default {
  name: 'ClientInbox',
  data() {
    return {
      messages: [],
    };
  },
  created() {
    this.$store.dispatch('getClientMessages').then(() => {
      this.messages = this.$store.getters.getClientMessages;
    });
  },
};
</script>

<style lang="scss">
.message-item {
  color: $dark;
  border: 2px solid #dedede;
  background-color: #f1f1f1;
  border-radius: 5px;
  padding: 10px;
  margin: 10px 0;
}
</style>
