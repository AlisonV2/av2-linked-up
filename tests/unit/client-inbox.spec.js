import { shallowMount } from '@vue/test-utils';
import ClientInbox from '@/components/clients/ClientInbox.vue';

/**
 * @module ClientInboxTest
 */
describe('ClientInbox.vue', () => {
  it('Check if row class is relevant', () => {
    const component = shallowMount(ClientInbox);
    const row = component.find('.client-inbox')
    expect(row.exists()).toBe(true);
  });
});
