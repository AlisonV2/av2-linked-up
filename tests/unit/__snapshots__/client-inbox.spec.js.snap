// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientInbox.vue ClientInbox Snapshot 1`] = `
<div
  class="row client-inbox"
>
  
  
</div>
`;
